_base/src/middleware.ts

```diff
@@ -49,7 +49,44 @@ export default withAuth(
       return NextResponse.redirect(new URL("/", req.url));
     }
 
-    return NextResponse.next();
+    const nonce = Buffer.from(crypto.randomUUID()).toString("base64");
+    const cspHeader = `
+      default-src 'self';
+      script-src 'self' 'unsafe-inline' 'unsafe-eval' https://connect-js.stripe.com/v1.0/connect.js https://js.stripe.com;
+      style-src 'self' 'unsafe-inline' https://fonts.googleapis.com;
+      img-src 'self' blob: data:;
+      font-src 'self' https://fonts.gstatic.com;
+      object-src 'none';
+      frame-src https://connect-js.stripe.com/ https://js.stripe.com;
+      base-uri 'self';
+      form-action 'self';
+      frame-ancestors 'none';
+      upgrade-insecure-requests;
+  `;
```

Comment:
I copied this from https://nextjs.org/docs/app/building-your-application/configuring/content-security-policy but had to modify it a few times to make things load.